### DISCUSSION TITLE
mincut in seconds for api v1

### DISCUSSION THREAD

**Discussion by idMysteries** [WyattBlue/auto-editor#515] [4. Bugs] [answered]
APIv1 does not support mincut in seconds, only in the number of frames. But at the same time, **margin in seconds works**.

If margin works in seconds, then, as it seems to me, it is possible to implement mincut in seconds.

```
Error! variable `timebase` not found. Did you mean a string literal.
 ```

```lua
local auto_editor_args = {
        "--export", "timeline:api=1",
        "--quiet",
        "--no-cache",
        "--progress", "none",
        "--margin", config.margin, -- "0.1s,0.2s"
        "--edit", string.format("audio:stream=%d,threshold=%s,mincut=1s", audio_stream_index - 1, config.threshold)
    }
```

Upd. I can convert seconds to frames in the script, which I did not think of.

**Comment by idMysteries** [comment]
auto-editor FILE --export timeline:api=1 --quiet --no-cache --progress none --edit audio:mincut=24,threshold=3%

```
"chunks": [
    [
      0,
      20,
      99999.0
    ],
```

Why do I have 20 frames of silence if mincut is 24?

**Comment by WyattBlue** [comment, marked as answer]
This is fixed in `master` / will be fixed in `25.1.1`

`margin` has a higher priority than `minclip` `mincut`

**Comment by idMysteries** [comment]
--export timeline:api=1 --quiet --no-cache --progress none --edit audio:mincut=1s,threshold=3%
```
[
      116,
      135,
      99999.0
    ],
```
Video fps: 23.976
Mincut 1s -> 19 frames of silence?


How does mincut and minclip work? How do I set up so that the minimum silence is as much in seconds as I want.
